test/gtest/uct/ib/test_ib.cc

```diff
@@ -259,27 +320,54 @@ class test_uct_ib : public uct_test {
 protected:
     entity *m_e1, *m_e2;
     static size_t ib_am_handler_counter;
+    struct ibv_device **device_list;
+    int num_devices;
 };
 
 size_t test_uct_ib::ib_am_handler_counter = 0;
 
-UCS_TEST_P(test_uct_ib, non_default_pkey, "IB_PKEY=0x2")
+UCS_TEST_P(test_uct_ib, non_default_pkey)
 {
-    ib_port_desc_t *port_desc;
+    /* test with invalid pkey set */
+    for (unsigned pkey = 0; pkey <= UCT_IB_PKEY_PARTITION_MASK; pkey++) {
+        modify_config("IB_PKEY", "0x" + ucs::to_hex_string(pkey));
 
-    /* check if the configured pkey exists in the port's pkey table.
-     * skip this test if it doesn't. */
-    port_desc = (ib_port_desc_t *) calloc(1, sizeof(*port_desc));
-    test_port_avail(port_desc);
+        ib_port_desc_t port_desc = { 0 };
+        test_port_avail(&port_desc);
 
-    if (port_desc->have_pkey) {
-        free(port_desc);
-    } else {
-        free(port_desc);
-        UCS_TEST_SKIP_R("pkey not found or not an IB port");
+        if (!port_desc.have_pkey) {
+            send_recv_short();
```

Comment:
need to run if have_pkey?

Reply:
the idea of this test is to set invalid pkey and try to initialize UCT - it has to be successful
so, we run this test in case of `have_pkey == false`